http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala

```diff
@@ -71,8 +71,11 @@ final class LeoAppServiceInterp[F[_]: Parallel](
       )
 
       samResourceId <- F.delay(AppSamResourceId(UUID.randomUUID().toString))
+
+      // Look up the original email in case this API was called by a pet SA:
+      originatingUserEmail <- authProvider.lookupOriginatingUserEmail(userInfo)
```

Comment:
We might want to do this further up the stack in the route layer, so it applies to all `/api` requests. Maybe in https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/StandardUserInfoDirectives.scala ?

Reply:
Actually thinking more… this is probably not needed for every request, because we already make Sam checks for those. Maybe we should just do this in places where we create a pet -- e.g. createApp and createRuntime.

Reply:
So just before we [create a UserInfo object on line 20 here](https://github.com/DataBiosphere/leonardo/blob/8972cd97e1df768eb9f07ab03c4afb97e4bda383/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/StandardUserInfoDirectives.scala#L20), we want to check that (1) the user is enabled and (2) if a pet SA was used to submit the request, we want to map to the user's email address and use that for all the requests moving forward?

Reply:
Yeah sorry I backtracked a bit there. :)

Sounds good re `createRuntime`. Technically deletes don't request pets so I don't _think_ it's needed there. But it also doesn't hurt so I don't feel too strongly.

Reply:
Oh I see. If I run delete app request through notebook, won't it pass the pet SA token? 🤔

Reply:
deleteApp won't request a new pet from Sam though. So it might just work as-is with the pet token. However I haven't tested it so not sure.

Reply:
Ah I see. I will test it out 👍

Reply:
yea, this feel right...in most cases, users are still calling the API as their original email